src/jit.rs

```diff
@@ -873,29 +876,24 @@ impl<'a, C: ContextObject> JitCompiler<'a, C> {
 
     #[inline]
     fn emit_sanitized_load_immediate(&mut self, destination: u8, value: i64) {
+        let lower_key = self.immediate_value_key as i32 as i64;
         if value >= i32::MIN as i64 && value <= i32::MAX as i64 {
-            let key = self.diversification_rng.gen::<i32>() as i64;
-            self.emit_ins(X86Instruction::load_immediate(destination, value.wrapping_sub(key)));
-            self.emit_ins(X86Instruction::alu(OperandSize::S64, 0x81, 0, destination, key, None));
+            self.emit_ins(X86Instruction::load_immediate(destination, value.wrapping_sub(lower_key)));
+            self.emit_ins(X86Instruction::alu(OperandSize::S64, 0x81, 0, destination, lower_key, None));
         } else if value as u64 & u32::MAX as u64 == 0 {
-            let key = self.diversification_rng.gen::<i32>() as i64;
-            self.emit_ins(X86Instruction::load_immediate(destination, value.rotate_right(32).wrapping_sub(key)));
-            self.emit_ins(X86Instruction::alu(OperandSize::S64, 0x81, 0, destination, key, None)); // wrapping_add(key)
+            self.emit_ins(X86Instruction::load_immediate(destination, value.rotate_right(32).wrapping_sub(lower_key)));
+            self.emit_ins(X86Instruction::alu(OperandSize::S64, 0x81, 0, destination, lower_key, None)); // wrapping_add(lower_key)
             self.emit_ins(X86Instruction::alu(OperandSize::S64, 0xc1, 4, destination, 32, None)); // shift_left(32)
+        } else if destination != REGISTER_SCRATCH {
+            self.emit_ins(X86Instruction::load_immediate(destination, value.wrapping_sub(self.immediate_value_key)));
+            self.emit_ins(X86Instruction::load_immediate(REGISTER_SCRATCH, self.immediate_value_key));
+            self.emit_ins(X86Instruction::alu(OperandSize::S64, 0x01, REGISTER_SCRATCH, destination, 0, None));
```

Comment:
I like the comments you add for clarity next to instructions. One could be useful here for signaling an addition.